packages/clerk-js/src/core/clerk.ts

```diff
@@ -209,6 +212,9 @@ export default class Clerk implements ClerkInterface {
 
   public openSignIn = (props?: SignInProps): void => {
     this.assertComponentsReady(this.#componentControls);
+    if (sessionExistsAndSingleSessionModeEnabled(this, this.#environment) && this.#instanceType === 'development') {
```

Comment:
Maybe extract the shared condition into a concrete method
```suggestion
    if (sessionExistsAndSingleSessionModeEnabled(this, this.#environment) && this.#isDevInstance() {
```

Reply:
We already use this logic in 4-5 other places so I'd prefer not to refactor all of these in this PR.